bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,387 @@
+        } catch (IOException e) {
+            logger.debug("Exception while trying to login: {}", e.getMessage(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Request new data and updates channels
+     */
+    private void updateData() {
+        updateDailyData();
+        updateMonthlyData();
+        updateYearlyData();
+    }
+
+    /**
+     * Request new dayly/weekly data and updates channels
+     */
+    private synchronized void updateDailyData() {
+        if (!isLinked(YESTERDAY) && !isLinked(LAST_WEEK) && !isLinked(THIS_WEEK)) {
+            return;
+        }
+
+        double lastWeek = -1;
+        double thisWeek = -1;
+        double yesterday = -1;
+        LinkyConsumptionData result = cachedDaylyData.getValue();
+        if (result != null && result.success()) {
+            LocalDate rangeStart = LocalDate.now().minusDays(13);
+            int jump = result.getDecalage();
+            while (rangeStart.getDayOfWeek() != weekFields.getFirstDayOfWeek()) {
+                rangeStart = rangeStart.plusDays(1);
+                jump++;
+            }
+
+            int lastWeekNumber = rangeStart.get(weekFields.weekOfWeekBasedYear());
+
+            lastWeek = 0;
+            thisWeek = 0;
+            yesterday = -1;
+            while (jump < result.getData().size()) {
+                double consumption = result.getData().get(jump).valeur;
+                if (consumption > 0) {
+                    if (rangeStart.get(weekFields.weekOfWeekBasedYear()) == lastWeekNumber) {
+                        lastWeek += consumption;
+                        logger.trace("Consumption at index {} added to last week: {}", jump, consumption);
+                    } else {
+                        thisWeek += consumption;
+                        logger.trace("Consumption at index {} added to current week: {}", jump, consumption);
+                    }
+                    yesterday = consumption;
+                }
+                jump++;
+                rangeStart = rangeStart.plusDays(1);
+            }
+        } else {
+            cachedDaylyData.invalidateValue();
+        }
+        updateKwhChannel(YESTERDAY, yesterday);
+        updateKwhChannel(THIS_WEEK, thisWeek);
+        updateKwhChannel(LAST_WEEK, lastWeek);
+    }
+
+    /**
+     * Request new monthly data and updates channels
+     */
+    private synchronized void updateMonthlyData() {
+        if (!isLinked(LAST_MONTH) && !isLinked(THIS_MONTH)) {
+            return;
+        }
+
+        double lastMonth = -1;
+        double thisMonth = -1;
+        LinkyConsumptionData result = cachedMonthlyData.getValue();
+        if (result != null && result.success()) {
+            int jump = result.getDecalage();
+            lastMonth = result.getData().get(jump++).valeur;
+            thisMonth = result.getData().get(jump).valeur;
```

Comment:
I think this is more understandable at a glance.
```suggestion
            lastMonth = result.getData().get(jump).valeur;
            thisMonth = result.getData().get(jump+1).valeur;
```

Reply:
I agree.